src/setup-beam.js

```diff
@@ -413,38 +413,37 @@ function getRunnerOSVersion() {
 }
 
 async function get(url0, pageIdxs) {
-  function getPage(pageIdx) {
+  function getPage(pageIdx, retries0) {
+    let retries = retries0
     return new Promise((resolve, reject) => {
+      if (retries === undefined) {
+        retries = 5
+      }
       const url = new URL(url0)
       if (pageIdx !== null) {
         url.searchParams.append('page', pageIdx)
       }
-      https
-        .get(
-          url,
-          {
-            headers: { 'user-agent': 'setup-beam' },
-          },
-          (res) => {
-            let data = ''
-            res.on('data', (chunk) => {
-              data += chunk
-            })
-            res.on('end', () => {
-              if (res.statusCode >= 400 && res.statusCode <= 599) {
-                reject(
-                  new Error(
-                    `Got ${res.statusCode} from ${url}. Exiting with error`,
-                  ),
-                )
-              } else {
-                resolve(data)
-              }
-            })
-          },
-        )
-        .on('error', (err) => {
-          reject(err)
+      request({
+        url: url.toString(),
+        headers: {
+          authorization: `Bearer ${process.env.GITHUB_TOKEN}`,
+          'user-agent': 'setup-beam',
+        },
+        fullResponse: false,
+        maxAttempts: 5,
+        retryDelay: 61250,
```

Comment:
We wait up until 61.25 seconds, on failure. This is because current rate limiting is something like 60 requests per hour. Maybe this is not even important (?)...

Reply:
@paulo-ferraz-oliveira WE can use the headers returned to figure out out when to retry. We may just want to use octokit rest package if it already does this though. 

I think we have to have caching to go along with it. I don't like any of these things FWIW 😁

Reply:
Never heard/used Octokit before, but can take a look.

Reply:
It might not support this, dunno. Either way, I think we need to get at doing this : 

https://docs.github.com/en/rest/overview/resources-in-the-rest-api?apiVersion=2022-11-28#conditional-requests

TL;DR we can hit a user's endpoint to see if the rate limit has already been hit, this doesn't count against the rate limit. Thus, we can be sure before we even fire off the first http call to the api for resources that we can actually do it. 

This makes all this more complicated, but it seems worth it. Thoughts?

Reply:
No longer valid.